shell/run_shell_cmd.go

```diff
@@ -57,7 +57,12 @@ func RunShellCommandWithOutput(terragruntOptions *options.TerragruntOptions, wor
 	// command requested by the user. The output of these other commands should not end up on stdout as this
 	// breaks scripts relying on terraform's output.
 	if !reflect.DeepEqual(terragruntOptions.TerraformCliArgs, args) {
-		outWriter = terragruntOptions.ErrWriter
+		errWriter = os.Stderr
+		outWriter = os.Stdout
+	}
+
+	if terragruntOptions.ErrWriter != os.Stderr {
+		errWriter = io.MultiWriter(terragruntOptions.ErrWriter, os.Stderr)
```

Comment:
Can you explain the thinking behind these changes a bit more?

Reply:
this is basically taking care of giving a summary of all errors at the end of a plan-all run and printing the errors while running... the check is needed to not print logs twice while running if used in a normal `plan` (run without being wrapped in a `plan-all`)..

Reply:
This is the `RunShellCommandWithOutput` method, which is used in many places in the code base, and not just during `plan-all` commands. I'm a little concerned about piling up workarounds within `RunShellCommandWithOutput` that are intended for other methods, as it's not at all obvious why this would be here from just reading this part of the code. Is there some other way to set `terragruntOptions` params correctly so this method does the right thing without one-off hacks?

Reply:
agreed.. ;) i can try to fix it in another way..

Reply:
as far as i have seen ErrWriter is not used in a consequent manner and fixing it in a nicer way would require a lot of refactoring as there are already a lot of exceptions throughout the code..